src/handshake.rs

```diff
@@ -148,7 +149,7 @@ fn generate_accept_key<'k>(key_base64: &WebSocketKey) -> [u8; 28] {
 	let d = digest.finalize();
 
 	let mut output_buf = [0; 28];
-	let n = base64::encode_config_slice(&d, base64::STANDARD, &mut output_buf);
+	let n = base64::engine::general_purpose::STANDARD.encode_slice(d, &mut output_buf).unwrap();
```

Comment:
Can you change this unwrap to expect with "encoding to base64 is exactly 28 bytes; qed" and remove the debug assert below?

Reply:
That wouldn't be equivalent. If it writes less than 28 bytes, it wouldn't error as it does currently.

I'm still fine doing it, I just want to confirm we're on the same page regarding that side effect;

Reply:
No. I'm referring to the fact the debug_assert will catch if base58 somehow encodes fewer bytes. Removing it for the unwrap/expect will only catch if base58 fails while trying to encode *more* bytes somehow.

Though all of this is such an unreachable edge case it's not worth too much of a discussion. I'll make the tweaks when I wake up :+1: